.eslintrc.json

```diff
@@ -1,43 +1,67 @@
 {
-	"env": {
+    "env": {
 		"browser": true,
-		"es2020": true,
+		"es2022": true,
+		"node": true,
 		"mocha": true
 	},
-	"extends": [
+    "extends": [
 		"eslint:recommended",
 		"plugin:@typescript-eslint/recommended",
+		"plugin:@typescript-eslint/recommended-requiring-type-checking",
 		"plugin:import/typescript"
 	],
-	"parser": "@typescript-eslint/parser",
-	"parserOptions": {
-		"ecmaVersion": 2020,
-		"sourceType": "script",
-		"ecmaFeatures": {
-			"impliedStrict": true
-		},
-		"project": "./tsconfig.json"
-	},
-	"plugins": [
-		"@typescript-eslint",
-		"eslint-plugin-import"
-	],
-	"rules": {
-		"@typescript-eslint/no-var-requires": "off", /** allows require statements outside of imports **/
+    "root": true,
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "ecmaVersion": 2022,
+        "sourceType": "commonjs",
+        "project": "./tsconfig.json"
+    },
+    "plugins": [
+        "@typescript-eslint"
+    ],
+    "rules": {
+        // eslint: recommended - disable core rules -> they're overridden by TS-eslint
+        "no-unused-vars": "off", // remove when fixed -> caused by eslint:recommended
+        "no-extra-semi": "off", // remove when fixed -> caused by eslint:recommended
+        "prefer-const": "warn",
+        "curly": "warn",
+        "eqeqeq": "warn",
+        "no-throw-literal": "warn",
 		"no-async-promise-executor": "off", /* Deactivated for now as i do not know how to fix it safely*/
-		"@typescript-eslint/no-namespace": "off",
-		"@typescript-eslint/no-explicit-any": "off",
-		"@typescript-eslint/no-floating-promises": "warn",
-		"import/no-extraneous-dependencies": [
-			"error",
-			{
-				"devDependencies": [
-					"**/test/**",
-					"**/ui-test/**"
-				],
-				"optionalDependencies": false,
-				"peerDependencies": false
-			}
-		]
```

Comment:
this o,e is very important to keep as it was gfollowing an ugly bug were the binary produced was not the one that was used by automated tests